src/main/java/com/microsoft/sqlserver/jdbc/ConfigRetryRule.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package com.microsoft.sqlserver.jdbc;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+
+
+/**
+ * The ConfigRetryRule object is what is used by the ConfigurableRetryLogic class to handle statement retries. Each
+ * ConfigRetryRule object allows for one rule.
+ *
+ */
+public class ConfigRetryRule {

```

Comment:
The naming is inconsistent with ConfigurableRetryLogic class. Should this be ConfigurableRetryRule?

Reply:
why does this need to be public?